monkey-runner/plugin/src/main/groovy/com/novoda/monkey/MonkeyConfigurationPlugin.groovy

```diff
@@ -37,60 +36,39 @@ public class MonkeyConfigurationPlugin implements Plugin<Project> {
         def android = project.extensions.findByName("android")
         android.command.devices().eachWithIndex { device, index ->
 
-            def showOverlayTask = project.task("showOverlayDevice${index}", type: NotificationBarOverlay) {
-                show = true
-                deviceId = device.id
-            }
-
             def monkeyTask = project.task("runMonkeyDevice${index}", type: TargetedMonkey) {
                 packageName = extension.packageNameFilter
                 events = extension.eventsCount
                 deviceId = device.id
                 logFileName = extension.logFileName
-                categories = ["android.intent.category.MONKEY"]
-            }
-
-            def hideOverlay = project.task("hideOverlayDevice${index}", type: NotificationBarOverlay) {
-                show = false
-                deviceId = device.id
+                categories = extension.categories
             }
 
             def uninstallApp = project.task("uninstallMonkeyDevice${index}", type: TargetedUninstall) {
                 packageName = extension.packageNameFilter
                 deviceId = device.id
             }
 
-            hideOverlay.dependsOn uninstallApp
-            monkeyTask.dependsOn showOverlayTask
-            monkeyTask.dependsOn extension.taskDependency
-            monkeyTask.finalizedBy hideOverlay
-            runMonkeyAllTask.dependsOn monkeyTask
-        }
-    }
-
-    static class MonkeyRunnerExtension {
-
-        String taskDependency
-        Integer eventsCount
-        String packageNameFilter
-        String logFileName
-
-        void setDefaultsForOptionalProperties() {
-            eventsCount = 50000
-            logFileName = 'monkey.log'
-        }
-
-        void ensureMandatoryPropertiesPresent() {
-            if (taskDependency == null) {
-                notifyMissingProperty('taskDependency')
+            if (extension.useMonkeyTrap) {
+                def showOverlayTask = project.task("showOverlayDevice${index}", type: NotificationBarOverlay) {
+                    show = true
+                    deviceId = device.id
+                }
+
+                def hideOverlay = project.task("hideOverlayDevice${index}", type: NotificationBarOverlay) {
+                    show = false
+                    deviceId = device.id
+                }
+
+                hideOverlay.dependsOn uninstallApp
+                monkeyTask.dependsOn showOverlayTask
+                monkeyTask.finalizedBy hideOverlay
```

Comment:
D'oh yes.

show it.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/novoda/spikes/pull/155/files/3b663693511796692c2acfd1ced059b41a319dfb#r78627340>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACjfG5Zu5HkC-cb_tSNlKM7cKpRT55RXks5qpvkQgaJpZM4J5wkM>
> .
>

-- 
Disclaimer: The information in this e-mail and any attachments is the 
property of Novoda Ltd and is confidential and may be legally privileged. 
This e-mail is intended solely for the person or organisation to which it 
is addressed. Any disclosure, copying or other use of the information by 
any person or organisation who is not the intended recipient is strictly 
prohibited and may be unlawful. If you have received this e-mail in error, 
please inform the sender immediately and delete/destroy this e-mail and any 
copies of it. Novoda Ltd has taken reasonable precautions to minimise the 
risk of any software viruses which may damage your systems, but we advise 
that you take the necessary steps to ensure that no virus contamination is 
suffered. Novoda Ltd does not accept any liability for any loss or damage 
caused by the transmission of any virus.

Novoda Ltd, Company No: 347444, Registered in Scotland Registered 
Office: C/O Alexander Sloan, 38 Cadogan Street, Glasgow, G2 7HF, Scotland.  VAT 
Registration Number  GB 984 2525 93